### ISSUE TITLE
Exception: SerializationBinder not found when mixing .NET Standard and .NET Framework

### ISSUE THREAD

**Issue by epsitec** [JamesNK/Newtonsoft.Json#1186]
Here is my scenario:

* I've built a class library `A`, based on .NET Standard 1.3, which references Newtonsoft.Json.
* I am referencing `A` from another class library `Test`, based on .NET 4.6.2, which references Newtonsoft.Json too.

In my tests, I do this:

```c#
var assembly = ... // assembly A
var types = assembly.DefinedTypes; // throws ReflectionTypeLoadException
```

When I execute the tests found in `Test`, I get a ReflectionTypeLoadException with this
information stored in `LoaderExceptions`:

> "Could not load type 'Newtonsoft.Json.SerializationBinder' from assembly 'Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'.":"Newtonsoft.Json.SerializationBinder"

Since project `Test` is referencing the full Newtonsoft.Json assembly first, the version which gets
loaded in memory does not contain the definition for `SerializationBinder`, which is available in
the .NET Standard version of Newtonsoft.Json.

Should the full-blown Newtonsoft.Json also provide an implementation of `SerializationBinder`
to avoid this kind of issues? Or am I missing something obvious here?

**Comment by epsitec** [comment]
I can work around the issue by forcing the `Test` assembly to use the DLL provided for `netstandard1.0`
by editing the `*.csproj` file manually:

```xml
    <Reference Include="Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
      <HintPath>..\..\packages\Newtonsoft.Json.9.0.1\lib\netstandard1.0\Newtonsoft.Json.dll</HintPath>
      <Private>True</Private>
    </Reference>
```

But this does not feel right.

**Comment by JamesNK** [comment]
Already done https://github.com/JamesNK/Newtonsoft.Json/commit/4f8a22594f4ee3dffcbb4a6e6cdbd8c064c86777

Might be in the pre-release version on NuGet, otherwise you'll have to wait.

**Comment by epsitec** [comment]
I've given Newtonsoft.Json 9.0.2-beta1 a try (which has been produced after #4f8a225) and there is still no implementation for `SerializationBinder` in the net45 version of Newtonsoft.Json, whereas the one for netstandard1.1 provides one. So my original issue subsists.

**Comment by epsitec** [comment]
@JamesNK I believe this issue should be reopened.

**Comment by JamesNK** [comment]
![image](https://cloud.githubusercontent.com/assets/303201/21972822/37ff5718-dc21-11e6-84ce-39e8b86a8238.png)

**Comment by epsitec** [comment]
![image](https://cloud.githubusercontent.com/assets/3872435/21972956/883e1d2c-dbbd-11e6-8f9a-875e67856ab4.png)

**Comment by epsitec** [comment]
The public members of Newtonsoft.Json is different for netstandard1.1 and net45.

**Comment by JamesNK** [comment]
And that is why everything that uses SerializationBinder was marked obsolete and ISerializationBinder was added. Adding Newtonsoft.Json.SerializationBinder to net45 and changing everything to use it iis a major breaking change and is not happening.

**Comment by epsitec** [comment]
How about making type `SerializationBinder` internal?

**Comment by JamesNK** [comment]
Breaking change.

**Comment by epsitec** [comment]
But in the end, you will need to implement that breaking change, or else we will have this incoherence between the public types, depending on what framework gets targeted. And that is no good either.

**Comment by JamesNK** [comment]
> I don't see any other solution...

I keep pointing it out to you: `ISerializationBinder`
